src/main/java/org/jfree/chart/plot/PolygonPlot.java

```diff
@@ -0,0 +1,137 @@
+package org.jfree.chart.plot;
+import java.awt.*;
+import javax.swing.JPanel;
+import org.jfree.chart.ChartFactory;
+
+import org.jfree.chart.axis.AxisLocation;
+import org.jfree.chart.swing.ChartPanel;
+import org.jfree.chart.JFreeChart;
+import org.jfree.chart.annotations.XYPolygonAnnotation;
+import org.jfree.chart.renderer.xy.XYLineAndShapeRenderer;
+import org.jfree.chart.swing.UIUtils;
+import org.jfree.data.xy.XYDataset;
+import org.jfree.chart.swing.ApplicationFrame;
+import org.jfree.chart.axis.NumberAxis;
+
+/**
+ *  CS427: Issue link: https://github.com/jfree/jfreechart/issues/234
+ * This demo illustrates the solution to issue #234.
+ */
+public class PolygonPlot extends ApplicationFrame{
+    private static XYDataset dataset = null;
+    private static XYPlot xyplot;
+
+    /**
+     * Creates a new demo instance.
+     *
+     * @param title  the frame title.
+     */
+    public PolygonPlot(final String title) {
+        super(title);
+        final JPanel chartPanel = createDemoPanel(title);
+        chartPanel.setPreferredSize(new Dimension(1000, 500));
+        setContentPane(chartPanel);
+    }
+
+    /**
+     * Creates a chart.
+     *
+     *
+     * @return The chart.
+     */
+
+    private static JFreeChart createChart(final String title) {
+        final JFreeChart chart = ChartFactory.createScatterPlot(
+                title, "Distance(m)", "Depth(m)", dataset,
+                PlotOrientation.VERTICAL, true, true, false);
+
+
+        xyplot = (XYPlot) chart.getPlot();
+        xyplot.setDomainPannable(true);
+        xyplot.setRangePannable(true);
+        xyplot.setDomainAxisLocation(0, AxisLocation.TOP_OR_LEFT);
+
+        final NumberAxis domain = (NumberAxis) xyplot.getDomainAxis();
+        domain.setRange(60.0, 800.0);
+
+
+        final NumberAxis range = (NumberAxis) xyplot.getRangeAxis();
+        range.setRange(0.0, 150.0);
+        range.setInverted(true);
+        return  chart;
+    }
+
+    public void paint(final Graphics graphics){
+        //Graphics2D g2d = (Graphics2D)g;
+        final XYLineAndShapeRenderer renderer
+                = (XYLineAndShapeRenderer) xyplot.getRenderer();
+        xyplot.setBackgroundPaint(Color.white);
+
+        final Color color1 = new Color(138, 43, 226);
+        final Color color2 = new Color(18, 10, 255);
+        final Color color3 = new Color(18, 255, 0);
+        final Color color4 = new Color(255, 255, 0);
+
+
+        final XYPolygonAnnotation polygon1 = new XYPolygonAnnotation(new double[] {70.0,
```

Comment:
You should not add annotations within the `paint()` method because this will be called many times and you will keep adding more and more annotations to the renderer.